### PR TITLE
User should be redirected to artwork page after confirmation that a bid is winning

### PR DESCRIPTION
**Pull request by yuki24**
addresses https://artsyproduct.atlassian.net/browse/AUCT-712

**Do not squash the commits as each commit has a meaningful message.** 

Adds polling logic that checks the bidding status and redirects the user to the artwork page when the user is winning.

### Placing a user as a registered user

AUCT-712

### 

Screen Shot 2019-11-06 at 12 17 05 PM

### CODE REVIEW

**Comment by yuki24** [comment]
I don't believe the peril comment is relevant as we auto-update the CHANGELOG upon each release.

File: src/Apps/Auction/Components/BidForm.tsx [hunk 1]
```diff
@@ -57,18 +63,38 @@ const getSelectedBid = ({
...
+  bidder: BidForm_saleArtwork["sale"]["registrationStatus"],
+  me: BidForm_me
+) => {
+  const isRegistered = !!bidder
+
+  return {
+    requiresCheckbox: !isRegistered,
+    requiresPaymentInformation: !(isRegistered || me.hasQualifiedCreditCards),
+  }
+}
```
Comment (erikdstock) [inline src/Apps/Auction/Components/BidForm.tsx]: Nice, I like capturing these points in one area.

File: src/Apps/Auction/Routes/ConfirmBid/index.tsx [hunk 2]
```diff
@@ -34,8 +37,12 @@ interface ConfirmBidProps {
   location: Location
 }
 
+const MAX_POLL_ATTEMPTS = 20
+
 export const ConfirmBidRoute: React.FC<ConfirmBidProps> = props => {
-  const { artwork } = props
+  let pollCount = 0
```
Comment (erikdstock) [inline src/Apps/Auction/Routes/ConfirmBid/index.tsx]: minor question here whether there is a better way to do this using a react hook.

**Comment by bhoggard** [comment]
I don't see a branch where we display the credit card form. Is that handled by a redirect case somewhere?

File: src/Apps/Auction/Routes/__tests__/ConfirmBid.test.tsx [hunk 3]
```diff
@@ -93,189 +102,220 @@ describe("Routes/ConfirmBid", () => {
...
-          quantity: 1,
-          price: 50000,
-        },
-      ],
+      mockBidderPositionQuery.mockReset()
+      mockBidderPositionQuery.mockResolvedValue(
+        confirmBidBidderPositionQueryWithWinning
+      )
+
+      setTimeout(() => {
```
Comment (yuki24) [inline src/Apps/Auction/Routes/__tests__/ConfirmBid.test.tsx]: This is a bit regrettable but our test setup is a bit heavier than I initially anticipated, so I couldn't figure out how to use the Jest timer in tandem with the page object. But there must be a better way than `setTimeout`.

cc @ds300

**Comment by yuki24** [comment]
I'm still a bit concerned about the `setTimeout`, but let's re-visit once we have a point-person who can look into it. Merging.

File: src/Apps/Auction/Components/BidForm.tsx (hunk 1 above)
Comment (dleve123) [reply src/Apps/Auction/Components/BidForm.tsx]: +1 – really readable code!

File: src/Apps/Auction/Routes/__tests__/ConfirmBid.test.tsx (hunk 3 above)
Comment (ds300) [reply src/Apps/Auction/Routes/__tests__/ConfirmBid.test.tsx]: I think it should be possible to use jest's mock timers with `createTestEnv`. Do you remember what issues you ran into? Happy to pair on this if you have some spare minutes soon!

**Pull request merged**